doc/compiler/optimizer/inlineFastPathLocations.md

```diff
@@ -0,0 +1,119 @@
+`java.lang.Math.abs()`. It can be converted into an `iabs` (absolute value of integer) [opcode](https://github.com/eclipse/omr/blob/e3a15a993c8aba80582aa1d6f3071e122acbd4c4/compiler/il/OMROpcodes.enum#L1911) .
+
+The call to `java.lang.Math.abs()` can be eliminated because there is a better representation of the execution in IL which
+is cheaper. That is a basic pattern of taking a call, recognizing them in a certain way, and optimizing it, sometimes using
+opcode, sometimes using a sequence of IL trees etc.
+
+
+There are multiple locations in our code where we do inline fast path optimizations:
+
+1. [Inliner](https://github.com/eclipse/omr/blob/master/compiler/optimizer/Inliner.cpp)
+2. [RecognizedCallTransformer](https://github.com/eclipse/omr/blob/master/compiler/optimizer/OMRRecognizedCallTransformer.cpp)
+3. [IL Generator](https://github.com/eclipse-openj9/openj9/tree/master/runtime/compiler/ilgen)
+4. [UnsafeFastPath](https://github.com/eclipse-openj9/openj9/blob/master/runtime/compiler/optimizer/UnsafeFastPath.cpp)
+
+## RecognizedCallTransformer
+
+`RecognizedCallTransformer` is similar to what the Inliner does with respect to how it recognizes calls and converts them
+into sequences of IL trees. However. Inliner is more complicated. There are a lot of things going on with respect to heuristics.
+Inliner constantly checks: Do I have a budget and can I inline this call? There may be checks that prevent us from inlining
+a particular method into another one because too many methods have been inlined already.
+
+`RecognizedCallTransformer` does not need to worry about budget. It is much simpler. It does not have any particular heuristics.
+It performs transformations unconditionally because 99% of the chance it is beneficial to do so.
+
+`RecognizedCallTransformer` is run as part of `ilgenStrategyOpts`. [ilgenStrategyOpts](https://github.com/eclipse/omr/blob/e3a15a993c8aba80582aa1d6f3071e122acbd4c4/compiler/optimizer/OMROptimizer.cpp#L535-L544)
+is run as soon as IL is generated. It also runs on the generated IL for the callee that will be inlined before the callee
+is inlined into the caller.
+
+`RecognizedCallTransformer` is viewed as a minor cleanup or a basic simplification path. It converts calls into better
+performing or simpler IL trees. Since it runs early on, the later passes do not even see the call that is converted.
+
+Over the time, the intent may have been to take some of the other fast paths that happen inside the Inliner and the
+recognized methods into `RecognizedCallTransformer`. The majority of them have not been done. Only a few methods are moved
+to `RecognizedCallTransformer` to avoid the complexities of doing unsafe transformations in the middle of an already
+complex transformation to inline calls. It also has the added benefit of avoiding falling victim to the inliner heuristics
+and somehow end up not inlining the call.
+
```

Comment:
This last sentence seems a bit repetitive given what was said earlier ?

Reply:
The last sentence will be removed in the next commit

Reply:
addressed in 190c736